core/lib/mno_enterprise/concerns/models/ability.rb

```diff
@@ -19,7 +19,52 @@ module ClassMethods
   #==================================================================
   # Instance methods
   #==================================================================
-  def initialize(user)
+  def initialize(user, session)
+
+    #===================================================
+    # Authorizations for users using the company view
+    #===================================================
+    company_view_abilities(user)
+
+    #===================================================
+    # Impac
+    #===================================================
+    impac_abilities(user)
+
+    #===================================================
+    # Admin abilities
+    #===================================================
+    admin_abilities(user, session)
+
+    # Define abilities for the passed in user here. For example:
+    #
+    #   user ||= User.new # guest user (not logged in)
+    #   if user.admin?
+    #     can :manage, :all
+    #   else
+    #     can :read, :all
+    #   end
+    #
+    # The first argument to `can` is the action you are giving the user
+    # permission to do.
+    # If you pass :manage it will apply to every action. Other common actions
+    # here are :read, :create, :update and :destroy.
+    #
+    # The second argument is the resource the user can perform the action on.
+    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, :published => true
+    #
+    # See the wiki for details:
+    # https://github.com/CanCanCommunity/cancancan/wiki/Defining-Abilities
+  end
+
+  def company_view_abilities(user)
```

Comment:
I think the `user ||= MnoEnterprise::User.new(id: nil)` line should stay in the `initialize` method

Reply:
yep agreed.